functions/bind-db.sh

```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+source functions/base.sh
+
+get_db_vcap_service() {
+  local vcap_services="${1}"
+  local binding_name="${2:-null}"
+
+  if [[ -z "${binding_name}" ]] || [[ "${binding_name}" == "null" ]]
+  then
+    jq '[.[][] | select(.credentials.uri) | select(.credentials.uri | split(":")[0] == ("mysql","postgres","postgresql"))] | first | select (.!=null)' <<<"${vcap_services}"
+  else
+    get_binding_service "${vcap_services}" "${binding_name}"
+  fi
+}
+
+get_db_vcap_service_type() {
+  local db="${1}"
+  db_type="$(jq -r -e '.tags[] | select((.=="mysql") or (.=="postgres") or (.=="postgresql"))' <<<"${db}")" ||
+    db_type="$(jq -r -e '.credentials.uri | split(":")[0]' <<<"${db}")" ||
+    db_type=""
+
+  if [[ $db_type == "postgresql" ]]
+  then
+    db_type="postgres"
+  fi
+  echo $db_type
+}
+
+get_db_user() {
+  local db="${1}"
+  db_user=$(jq -r -e '.credentials.Username' <<<"${db}") ||
+    db_user=$(jq -r -e '.credentials.username' <<<"${db}") ||
+      db_user=$(jq -r -e '.credentials.uri |
+          split("://")[1] | split(":")[0]' <<<"${db}") ||
+        db_user=''
+  echo "${db_user}"
+}
+
+get_db_password() {
+  local db="${1}"
+  db_pass=$(jq -r -e '.credentials.Password' <<<"${db}") ||
+    db_pass=$(jq -r -e '.credentials.password' <<<"${db}") ||
+      db_pass=$(jq -r -e '.credentials.uri |
+          split("://")[1] | split(":")[1] |
+          split("@")[0]' <<<"${db}") ||
+        db_pass=''
+
+  echo "${db_pass}"
+}
+
+get_db_host() {
+  local db="${1}"
+  db_host=$(jq -r -e '.credentials.host' <<<"${db}") ||
+    db_host=$(jq -r -e '.credentials.hostname' <<<"${db}") ||
+      db_host=$(jq -r -e '.credentials.uri |
+          split("://")[1] | split(":")[1] |
+          split("@")[1] |
+          split("/")[0]' <<<"${db}") ||
+        db_host=$(jq -r -e '.credentials.uri |
+            split("://")[1] |
+            split("/")[0]' <<<"${db}") ||
+          db_host=''
+  echo "${db_host}"
+}
+
+get_db_name() {
+  local db="${1}"
+  db_name=$(jq -r -e '.credentials.database_name' <<<"${db}") ||
+    db_name=$(jq -r -e '.credentials.database' <<<"${db}") ||
+      db_name=$(jq -r -e '.credentials.uri |
+          split("://")[1] | split("/")[1] |
+          split("?")[0]' <<<"${db}") ||
+        db_name=''
+  echo "${db_name}"
+}
+
+get_db_port() {
+  local db_type=${1}
+  local db_port=""
+
+  if [[ "${db_type}" == "mysql" ]]
+  then
+      db_port="3306"
```

Comment:
Could pull out of the binding ENV var?

Although 3306 is the default MySQL port, and it is currently pinned and non configurable via CSB